### PR TITLE
Make Docker test robust when removing a temp dir

### LINKED ISSUE #61553: [CI] DockerTests.test071BindMountCustomPathWithDifferentUID
**Build scan**:https://gradle-enterprise.elastic.co/s/zytljvsexacbm

**Repro line**: none logged 

**Reproduces locally?**: no idea

**Applicable branches**: 7.x, master

**Failure history**:
https://build-stats.elastic.co/goto/9305134dac50d3ef2708a0d2cf279f57

**Failure excerpt**:

```
:qa:os:destructiveDistroTest.default-docker_ubi FAILED

org.elasticsearch.packaging.test.DockerTests > test071BindMountCustomPathWithDifferentUID FAILED
    org.elasticsearch.packaging.util.Shell$ShellException: Command was not successful: [bash -c docker run --rm --volume "/tmp/DockerTests13638892981020840716:/mount" alpine /bin/sh -c "cd /mount && rm -r esDataDir"]
       result: exitCode = [1] stdout = [] stderr = [rm: can't remove 'esDataDir/_state/pending_segments_e': No such file or directory]
        at __randomizedtesting.SeedInfo.seed([33BF03973DECA4FB:7EF34C50A75223C3]:0)
        at org.elasticsearch.packaging.util.Shell.runScript(Shell.java:158)
        at org.elasticsearch.packaging.util.Shell.run(Shell.java:84)
        at org.elasticsearch.packaging.util.Docker.executePrivilegeEscalatedShellCmd(Docker.java:409)
        at org.elasticsearch.packaging.util.Docker.rmDirWithPrivilegeEscalation(Docker.java:451)
        at org.elasticsearch.packaging.test.DockerTests.lambda$test071BindMountCustomPathWithDifferentUID$1(DockerTests.java:217)
        at org.elasticsearch.packaging.util.Platforms.onLinux(Platforms.java:74)
        at org.elasticsearch.packaging.test.DockerTests.test071BindMountCustomPathWithDifferentUID(DockerTests.java:199)


Suite: Test class org.elasticsearch.packaging.test.DockerTests
  2> org.elasticsearch.packaging.util.Shell$ShellException: Command was not successful: [bash -c docker run --rm --volume "/tmp/DockerTests13638892981020840716:/mount" alpine /bin/sh -c "cd /mount && rm -r esDataDir"]
       result: exitCode = [1] stdout = [] stderr = [rm: can't remove 'esDataDir/_state/pending_segments_e': No such file or directory]
        at __randomizedtesting.SeedInfo.seed([33BF03973DECA4FB:7EF34C50A75223C3]:0)
        at org.elasticsearch.packaging.util.Shell.runScript(Shell.java:158)
        at org.elasticsearch.packaging.util.Shell.run(Shell.java:84)
        at org.elasticsearch.packaging.util.Docker.executePrivilegeEscalatedShellCmd(Docker.java:409)
        at org.elasticsearch.packaging.util.Docker.rmDirWithPrivilegeEscalation(Docker.java:451)
        at org.elasticsearch.packaging.test.DockerTests.lambda$test071BindMountCustomPathWithDifferentUID$1(DockerTests.java:217)
        at org.elasticsearch.packaging.util.Platforms.onLinux(Platforms.java:74)
        at org.elasticsearch.packaging.test.DockerTests.test071BindMountCustomPathWithDifferentUID(DockerTests.java:199)

Tests with failures:
 - org.elasticsearch.packaging.test.DockerTests.test071BindMountCustomPathWithDifferentUID
```

### LINKED ISSUE #61614: [CI] Packaging Test 071 BindMountCustomPathWithDifferentUID failure for UBI
**Build scan**: https://gradle-enterprise.elastic.co/s/rmxnfg5jcgyno/tests/:qa:os:destructiveDistroTest.default-docker_ubi/org.elasticsearch.packaging.test.DockerTests/test071BindMountCustomPathWithDifferentUID#1

**Repro line**: 
```
./gradlew :qa:os:destructiveDistroTest.default-docker_ubi
```

**Reproduces locally?**: N/A 

**Applicable branches**: master

**Failure excerpt**:
```
org.elasticsearch.packaging.util.Shell$ShellException: Command was not successful: [bash -c docker run --rm --volume "/tmp/DockerTests2807067604740472611:/mount" alpine /bin/sh -c "cd /mount && rm -r esDataDir"] 
  result: exitCode = [1] stdout = [] stderr = [rm: can't remove 'esDataDir/_state': Directory not empty]
```

Possibly related to #61194

### PR DESCRIPTION
**Pull request by pugnascotia**
Closes #61614. Closes #61553.

In a particular Docker test, ensure is shut down before removing the
data directory. Also add the force flag to `rm`.

### CODE REVIEW

File: qa/os/src/test/java/org/elasticsearch/packaging/test/DockerTests.java [hunk 1]
```diff
@@ -214,6 +214,10 @@ public void test071BindMountCustomPathWithDifferentUID() throws Exception {
             assertThat(nodes.at("/_nodes/successful").intValue(), equalTo(1));
             assertThat(nodes.at("/_nodes/failed").intValue(), equalTo(0));
 
+            // Ensure container is stopped before we remove tempEsDataDir, so ensure nothing
```
Comment (rjernst) [inline qa/os/src/test/java/org/elasticsearch/packaging/test/DockerTests.java:217]: so ensure -> so

**Review by rjernst — APPROVED** [review]
LGTM

**Comment by pugnascotia** [comment]
@elasticmachine update branch

**Comment by pugnascotia** [comment]
Backported to `7.x` in 50119bb08f1

**Pull request merged**